CRM/Upgrade/Incremental/php/FiveSeventySix.php

```diff
@@ -32,6 +34,69 @@ public function upgrade_5_76_alpha1($rev): void {
     $this->addTask('Add start_date to civicrm_mailing table', 'addColumn', 'civicrm_mailing', 'start_date', "timestamp NULL DEFAULT NULL COMMENT 'date on which this mailing was started.'");
     $this->addTask('Add end_date to civicrm_mailing table', 'addColumn', 'civicrm_mailing', 'end_date', "timestamp NULL DEFAULT NULL COMMENT 'date on which this mailing was completed.'");
     $this->addTask('Add status to civicrm_mailing table', 'addColumn', 'civicrm_mailing', 'status', "varchar(12) DEFAULT NULL COMMENT 'The status of this Mailing'");
+    $this->addTask('Update civicrm_mailing to permit deleting records from civicrm_mailing_job', 'updateNewCiviMailFields');
+  }
+
+  public static function updateNewCiviMailFields(CRM_Queue_TaskContext $ctx): bool {
+    [$minId, $maxId] = CRM_Core_DAO::executeQuery("SELECT coalesce(min(id),0), coalesce(max(id),0)
+      FROM civicrm_mailing ")->getDatabaseResult()->fetchRow();
+    for ($startId = $minId; $startId <= $maxId; $startId += self::MAILING_BATCH_SIZE) {
```

Comment:
@eileenmcnaughton I think for the upgrade tests we need to handle where $minId and $maxId == 0

Reply:
ah thanks